test/extended/oauth/groupsync.go

```diff
@@ -1,111 +1,419 @@
+				currentTmpFilePath := path.Join(currentTmpDirPath, testFileName)
 
-		// Copy groupsync script
-		err = pod.CopyFromHost(groupSyncScriptPath, path.Join("/usr", "bin", "groupsync.sh"))
-		o.Expect(err).NotTo(o.HaveOccurred())
+				if strings.Contains(testFileName, "sync-config") || strings.Contains(testFileName, "valid") {
+					// update sync-configs and validation files with the LDAP server's IP
+					fileContent = []byte(strings.ReplaceAll(string(fileContent), "LDAP_SERVICE_IP:389", ldapAddress))
+					fileContent = []byte(strings.ReplaceAll(string(fileContent), "LDAP_SERVICE_IP", "127.0.0.1"))
+					fileContent = []byte(strings.ReplaceAll(string(fileContent), "LDAP_CA", ldapCAPath))
+				}
 
-		// Fix flake executing groupsync.sh before it has landed on the pod.
-		err = wait.PollImmediate(2*time.Second, 5*time.Minute, func() (done bool, err error) {
-			_, lsErr := pod.Exec("/bin/ls /usr/bin/groupsync.sh &> /dev/null")
-			if lsErr != nil {
-				e2e.Logf("groupsync.sh is not available, retrying...")
-				return false, nil
+				err = ioutil.WriteFile(currentTmpFilePath, fileContent, 0644)
+				o.Expect(err).NotTo(o.HaveOccurred())
 			}
-			return true, nil
-		})
-		o.Expect(err).NotTo(o.HaveOccurred())
+		}
 
-		// Make it executable
-		_, err = pod.Exec("chmod +x /usr/bin/groupsync.sh")
-		o.Expect(err).NotTo(o.HaveOccurred())
+		userClient := oc.AdminUserClient()
 
-		// Execute groupsync.sh
-		_, err = pod.Exec(fmt.Sprintf("export LDAP_SERVICE=%s LDAP_CA=%s ADMIN_KUBECONFIG=%s; groupsync.sh",
-			ldapService, path.Join(remoteTmp, caFileName), path.Join(remoteTmp, kubeConfigFileName)))
-		o.Expect(err).NotTo(o.HaveOccurred())
+		for _, schema := range []string{"rfc2307", "ad", "augmented-ad"} {
+			currentDir := path.Join(tmpDir, schema)
+
+			// load OpenShift and LDAP group UIDs, needed for literal whitelists
+			groupUIDContents, err := ioutil.ReadFile(path.Join(currentDir, "ldapgroupuids.txt"))
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			osGroupUIDContents, err := ioutil.ReadFile(path.Join(currentDir, "osgroupuids.txt"))
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			groupUIDs := strings.Split(string(groupUIDContents), "\n")
+			osGroupUIDs := strings.Split(string(osGroupUIDContents), "\n")
+
+			o.Expect(len(groupUIDs)).To(o.Equal(3))
+			o.Expect(len(osGroupUIDs)).To(o.Equal(3))
+
+			// define paths used in the test
+			syncConfigPath := path.Join(currentDir, "sync-config.yaml")
+			syncConfigUserDefinedPath := path.Join(currentDir, "sync-config-user-defined.yaml")
+			whitelistOpenshiftPath := path.Join(currentDir, "whitelist_openshift.txt")
+			whitelistLdapPath := path.Join(currentDir, "whitelist_ldap.txt")
+			blackListOpenshiftPath := path.Join(currentDir, "blacklist_openshift.txt")
+			blacklistLdapPath := path.Join(currentDir, "blacklist_ldap.txt")
+
+			g.By("Sync all LDAP groups from LDAP server")
+			_, err = oc.Run("adm", "groups", "sync").Args("--sync-config="+syncConfigPath, "--confirm").Output()
```

Comment:
Use `.Execute()` instead of `.Output()` to only get error and ignore the rest :wink: